ui/src/screens/Jobs/JobsScreen.tsx

```diff
@@ -65,20 +66,26 @@ const JobsScreen: React.FC = function () {
       </Helmet>
       <Search placeholder="Search job postings" onChange={onSearchChange} />
       <CategorySelector category="jobs" />
-      <PostsContainer>
-        {state.jobs.map((post: PostInterface) => (
-          <Post category="jobs" active={post.active} key={post.id} data={post} />
-        ))}
-        <PostLoader hasMore={state.more} onLoadMore={handleLoadMore} />
-      </PostsContainer>
+      { state.fetched ? (
+        <PostsContainer>
```

Comment:
Not too sure if I can wrap it like this instead 
```
<PostsContainer>
{ state.fetched ? (
          {state.jobs.map((post: PostInterface) => (
            <Post category="jobs" active={post.active} key={post.id} data={post} />
          ))}
          <PostLoader hasMore={state.more} onLoadMore={handleLoadMore} />
   ) : (
   <PostSpinner /> 
   )
</PostsContainer>
```

without React complaining that a root element is needed. Using 
```
<>
          {state.jobs.map((post: PostInterface) => (
            <Post category="jobs" active={post.active} key={post.id} data={post} />
          ))}
          <PostLoader hasMore={state.more} onLoadMore={handleLoadMore} />
</>
```
works, however, will need to update the CSS in the `PostsContainer` component

If anyone knows how please let me know 🙇

Reply:
@jacobtyq The first example is broken code though, that's why eslint is complaining. Notice line 2-3:
```
{ state.fetched ? (
          {state.jobs.map((post: PostInterface) => (
```
You lifted `{state.jobs.map...` from it's previous position in "HTML" code into the current place which is "JS" code. So JS is confused by that opening `{` here: it thinks it's an object which doesn't make sense.

The opening `{` in `{state.jobs.map..` is not needed because it exists to indicate HTML crossing into JS boundary. In the new place, it is already in JS territory. Here's a working example to illustrate my meaning:
```
{ state.fetched ? (
    state.jobs.map((post: PostInterface) => (
        <Post category="jobs" active={post.active} key={post.id} data={post} />
    )
)...
```

**But**... anyway, you have two components here (including `<PostLoader>`), and you need the fragment `<>` anyway if you return more than one component. So your second snippet is completely correct. Putting the fragment means it's back in "HTML" mode, so the `{` can remain this time.